.github/workflows/integration-tests.yml

```
@@ -0,0 +1,60 @@
+      - name: Build & Up Docker image
+        run: |
+          docker compose -f tests/integration_tests/docker-compose-integration.yml build
+          docker compose -f tests/integration_tests/docker-compose-integration.yml up -d
+
+      - name: Run integration tests
+        run: |
+          docker compose -f tests/integration_tests/docker-compose-integration.yml run --rm integration_test pytest tests/integration_tests -m integration -v --tb=short --junit-xml=test-reports/integration-report.xml
+
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.8)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)